internal/db/update/schema.go

```diff
@@ -66,15 +78,17 @@ func (s *SchemaUpdate) Ensure(db *sql.DB) (int, error) {
 			return fmt.Errorf("failed to check if schema table is there: %w", err)
 		}
 
-		current := 0
+		var versions []int
 		if exists {
-			versions, err := query.SelectIntegers(ctx, tx, "SELECT version FROM schemas ORDER BY version")
+			// updateFromV1 changes the schema table and needs to be run before we calculate the schema version.
+			err := updateFromV1(ctx, tx)
 			if err != nil {
 				return err
 			}
 
-			if len(versions) > 0 {
-				current = versions[len(versions)-1]
+			versions, err = query.SelectIntegers(ctx, tx, "SELECT COALESCE(MAX(version), 0) FROM schemas WHERE type = 0 UNION ALL SELECT COALESCE(MAX(version), 0) FROM schemas WHERE type = 1")
```

Comment:
One last thing (sorry). Each time I come back to this PR I need to remember/figure out why `COALESCE(MAX(version), 0)` is required. 

Could you please define this query as a const with a description of the format of the schema table and reasoning as to why this query is the best way to extract the details we want. For example, why can't we just count the rows where `type = 0` and then count the rows where `type = 1`.

Reply:
Sounds good @markylaing - its good to improve readability for other people.

Reply:
Done.